returnn/tf/layers/basic.py

```diff
@@ -6485,6 +6485,98 @@ def get_sub_layers(self):
     return layers
 
 
+class SearchSortedLayer(LayerBase):
+  """
+  Basically wraps :func:`tf.searchsorted`.
+
+  Takes a tensor `sorted_sequence` that is sorted along one axis, and a tensor `values`.
+  Will compute an output tensor with the same axes as `values`,
+  where each entry is the index of the value within the sorted sequence.
+  All (batch) axes of `sorted_sequence` except for the axis it is sorted along must be present in `values`.
+  """
+  layer_class = "search_sorted"
+  recurrent = True  # do not shuffle the sorted_sequence
+
+  def __init__(self, sorted_sequence, values, axis="T", side="left", **kwargs):
+    """
+    :param LayerBase sorted_sequence:
+    :param LayerBase values: search values
+    :param str axis: the axis along which `sorted_sequence` is sorted
+    :param str side: "left" or "right".
+      When one of the `values` exactly matches an element of the `sorted_sequence`,
+      whether to choose the lower or higher index.
+    """
+    super(SearchSortedLayer, self).__init__(**kwargs)
+    self.sorted_sequence = sorted_sequence  # e.g. [B,T]
+    self.values = values  # e.g. [B',F']
```

Comment:
I think `B'` is misleading in the comments, because the batch axis has to be exactly the same for both inputs.

In my understanding I would say just `B` for both to say it has to be the same axis.

Reply:
Hm yes, I see. `B` here is just an example, it could be an entire set of common axes (maybe permuted differently, and at different axes).
Is it clearer like I have it now?